src/mscorlib/src/System/Reflection/Emit/CustomAttributeBuilder.cs

```diff
@@ -70,8 +70,14 @@ public CustomAttributeBuilder(ConstructorInfo con, Object[] constructorArgs)
         // Check that a type is suitable for use in a custom attribute.
         private bool ValidateType(Type t)
         {
-            if (t.IsPrimitive || t == typeof(String) || t == typeof(Type))
+            if (t.IsPrimitive)
+            {
+                return t != typeof(IntPtr) && t != typeof(UIntPtr);
```

Comment:
Hmm... on further look, the IsEnum logic (quite correctly) excludes chars and booleans. Probably not a good idea to unify them after all. Forget about this one.